pkg/azure/client/vmss.go

```diff
@@ -72,6 +72,9 @@ func (c VmssClient) Create(ctx context.Context, resourceGroupName, name string,
 func (c VmssClient) Delete(ctx context.Context, resourceGroupName, name string) error {
 	future, err := c.client.Delete(ctx, resourceGroupName, name)
 	if err != nil {
+		if IsAzureAPINotFoundError(err) {
+			return nil
+		}
```

Comment:
Is it possible to add a unit test to make sure that we won't regress this in the future?

Reply:
For the `IsAzureAPINotFoundError` function exists unit tests. 
The Azure api return a 404 in case the resource group is gone.

Reply:
Added a test for the case that the resource group gone. Sorry for the delay.